src/e2e/puppeteer/__tests__/caret-multiline.ts

```diff
@@ -0,0 +1,254 @@
+    )
+
+    const offset = await getSelection().focusOffset
+    expect(offset).toBeGreaterThan(70)
+  })
+
+  it('on cursorDown, the caret should move from near the end of a multi-line cursor to the beginning of a new cursor.', async () => {
+    const multiLineCursor =
+      "Beautiful antique furnishings fill this quiet, comfortable flat across from the Acropolis museum. AC works great. It is in an heavily touristic area, but the convenience can't be beat. Highly recommended."
+    const importText = `
+  - a
+    - b
+      - c
+      - ${multiLineCursor}
+      - d`
+
+    await paste(importText)
+
+    const editableNodeHandle = await waitForEditable(multiLineCursor)
+    await click(editableNodeHandle, { offset: multiLineCursor.length - 2 })
+
+    await press('ArrowDown')
+
+    // the focus must be at the beginning of 'd' after cursor down
+    const textContext = await getSelection().focusNode?.textContent
+    expect(textContext).toBe('d')
+
+    const offset = await getSelection().focusOffset
+    expect(offset).toBe(0)
+  })
+
+  it('on cursorUp, the caret should move from near the beginning of a multi-line cursor to the beginning of the previous cursor.', async () => {
+    const multiLineCursor =
+      "Beautiful antique furnishings fill this quiet, comfortable flat across from the Acropolis museum. AC works great. It is in an heavily touristic area, but the convenience can't be beat. Highly recommended."
+    const importText = `
+  - a
+    - b
+      - c
+      - ${multiLineCursor}
+      - d`
+
+    await paste(importText)
+
+    const editableNodeHandle = await waitForEditable(multiLineCursor)
+    await click(editableNodeHandle, { offset: 1 })
+
+    await press('ArrowUp')
+
+    // the focus must be at the beginning of 'c' after cursor up
+    const textContext = await getSelection().focusNode?.textContent
+    expect(textContext).toBe('c')
+
+    const offset = await getSelection().focusOffset
+    expect(offset).toBe(0)
+  })
+
+  it('on cursorUp, the caret should move from near the end of multi-line cursor to the previous line in the same cursor.', async () => {
+    const multiLineCursor =
+      "Beautiful antique furnishings fill this quiet, comfortable flat across from the Acropolis museum. AC works great. It is in an heavily touristic area, but the convenience can't be beat. Highly recommended."
+    const importText = `
+  - a
+    - b
+      - c
+      - ${multiLineCursor}
+      - d`
+
+    await paste(importText)
+
+    const editableNodeHandle = await waitForEditable(multiLineCursor)
+    await click(editableNodeHandle, { offset: multiLineCursor.length - 2 })
+
+    await press('ArrowUp')
+
+    // the focus must be in the middle of the multi-line cursor after cursor up
+    const textContext = await getSelection().focusNode?.textContent
+    expect(textContext).toBe(multiLineCursor)
+
+    const offset = await getSelection().focusOffset
+    expect(offset).toBeGreaterThan(70)
```

Comment:
Similar to the above, this will depend on the browser width.

This corresponds to test case (9).